File: NumberTheory/EuclideAlgorithm/euclide.cpp
#include<iostream>
using namespace std;

int gcd(int a, int b) {
    if (a < b) {
        return gcd(b,a);
    }

    if (b == 0) return a;

    return gcd(b, a%b);
}

int main() {
    int a,b;
    cin >> a >> b;
    cout << gcd(a,b) << endl;
}

File: NumberTheory/EuclideAlgorithm/euclide.js
(function() {
    (function main() {

        function gcd(a,b) {
            if (a < b) {
                return gcd(b,a);
            }
            if (b === 0) return a;
            return gcd(b, a%b);
        }

        console.log(gcd(15,5));
        console.log(gcd(20,5));
        console.log(gcd(96,14));
    }());
}());